homeassistant/components/vevor_heater/coordinator.py

```diff
@@ -0,0 +1,66 @@
+"""Data update coordinators for Vevor BLE Heater integration."""
+
+import logging
+from typing import Optional
+
+from bleak import BLEDevice
+from vevor_heater_ble.heater import VevorDevice
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DEFAULT_UPDATE_INTERVAL, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class VevorHeaterUpdateCoordinator(DataUpdateCoordinator[VevorDevice]):
+    """Update coordinator for Vevor BLE Heaters."""
+
+    def __init__(self, hass: HomeAssistant, device: BLEDevice) -> None:
+        """Initialize the coordinator for the given BLE device. Assumes the device corresponds to a Vevor Heater."""
+        self._ble_device = device
+        self._vevor_device = VevorDevice(
+            address=self._ble_device.address,
+            name=f"Vevor {self._ble_device.name or ''}",
```

Comment:
maybe use short address from `bluetooth_data_tools` if the name is unavailable

Reply:
Done